google-cloud-storage/src/test/java/com/google/cloud/storage/ITOpenTelemetryTest.java

```diff
@@ -58,29 +48,80 @@ public void checkInstrumentation() {
     StorageOptions storageOptions =
         StorageOptions.http().setOpenTelemetrySdk(openTelemetrySdk).build();
     Storage storage = storageOptions.getService();
-    storage.create(BucketInfo.of(generator.randomBucketName()));
+    String bucket = randomBucketName();
+    storage.create(BucketInfo.of(bucket));
     TestExporter testExported = (TestExporter) exporter;
     SpanData spanData = testExported.getExportedSpans().get(0);
     Assert.assertEquals("Storage", getAttributeValue(spanData, "gcp.client.service"));
     Assert.assertEquals("googleapis/java-storage", getAttributeValue(spanData, "gcp.client.repo"));
     Assert.assertEquals(
         "com.google.cloud.google-cloud-storage",
         getAttributeValue(spanData, "gcp.client.artifact"));
+    Assert.assertEquals("http", getAttributeValue(spanData, "rpc.system"));
+
+    // Cleanup
+    RemoteStorageHelper.forceDelete(storage, bucket);
   }
 
   @Test
-  public void noOpDoesNothing() {
-    StorageOptions storageOptions = StorageOptions.http().build();
+  public void checkInstrumentationGrpc() {
+    SpanExporter exporter = new TestExporter();
+
+    OpenTelemetrySdk openTelemetrySdk =
+        OpenTelemetrySdk.builder()
+            .setTracerProvider(
+                SdkTracerProvider.builder()
+                    .addSpanProcessor(SimpleSpanProcessor.create(exporter))
+                    .build())
+            .build();
+    StorageOptions storageOptions =
+        StorageOptions.grpc().setOpenTelemetrySdk(openTelemetrySdk).build();
     Storage storage = storageOptions.getService();
-    storage.create(BucketInfo.of(generator.randomBucketName()));
+    String bucket = randomBucketName();
+    storage.create(BucketInfo.of(bucket));
+    TestExporter testExported = (TestExporter) exporter;
+    SpanData spanData = testExported.getExportedSpans().get(0);
+    Assert.assertEquals("Storage", getAttributeValue(spanData, "gcp.client.service"));
+    Assert.assertEquals("googleapis/java-storage", getAttributeValue(spanData, "gcp.client.repo"));
+    Assert.assertEquals(
+        "com.google.cloud.google-cloud-storage",
+        getAttributeValue(spanData, "gcp.client.artifact"));
+    Assert.assertEquals("grpc", getAttributeValue(spanData, "rpc.system"));
+
+    // Cleanup
+    RemoteStorageHelper.forceDelete(storage, bucket);
+  }
+
+  @Test
+  public void noOpDoesNothing() {
+    String httpBucket = randomBucketName();
+    String grpcBucket = randomBucketName();
+    // NoOp for HTTP
+    StorageOptions storageOptionsHttp = StorageOptions.http().build();
+    Storage storageHttp = storageOptionsHttp.getService();
+    storageHttp.create(BucketInfo.of(httpBucket));
+
+    // NoOp for Grpc
+    StorageOptions storageOptionsGrpc = StorageOptions.grpc().build();
+    Storage storageGrpc = storageOptionsGrpc.getService();
+    storageGrpc.create(BucketInfo.of(grpcBucket));
```

Comment:
What is no-op testing and checking?

Reply:
just checking to make sure that it still runs without issue and does nothing. basically just that it doesnt error out.